tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ScalaJSClassEmitter.scala

```diff
@@ -478,6 +547,24 @@ private[scalajs] final class ScalaJSClassEmitter(
     }
   }
 
+  private[tools] def genIntervalsTest(className: String, tag: js.Tree)
+      (implicit pos: Position): js.Tree = {
+    import TreeDSL._
+
+    val intervals = subtypeIntervals(className)
+
+    if(intervals.nonEmpty && !needsSubtypeArray(className)){
+      intervals.map {
+        case (a, b) if a == b => tag === js.IntLiteral(a)
+        case (lo, hi) =>
+          js.BinaryOp(JSBinaryOp.>=, tag, js.IntLiteral(lo)) &&
+          js.BinaryOp(JSBinaryOp.<=, tag, js.IntLiteral(hi))
+      } reduce(_ || _)
```

Comment:
`}.reduce` with a `.`. We don't use infix syntax on alphanumeric methods anymore.